lib/presentation/widget/myspace/my_space_widget.dart

```diff
@@ -136,11 +136,13 @@ class _MySpaceWidgetState extends State<MySpaceWidget> {
           if (documentList.isEmpty) {
             return _buildUploadFileHere(context);
           } else {
-            return ListView.builder(
+            return RefreshIndicator(
```

Comment:
How about like this: 
```
Widget _buildMySpaceList(BuildContext context) {
    return StoreConnector<AppState, MySpaceState>(
        converter: (store) => store.state.mySpaceState,
        builder: (context, mySpaceState) {
          return mySpaceState.viewState.fold(
              (failure) => _buildMySpaceListView(context, mySpaceState.documentList),
              (success) =>
                  success is LoadingState ?
                  _buildMySpaceListView(context, mySpaceState.documentList) :
                  RefreshIndicator(
                      onRefresh: () async => mySpaceViewModel.getAllDocument(),
                      child: _buildMySpaceListView(context, mySpaceState.documentList)));
        });
  }

  Widget _buildMySpaceListView(
      BuildContext context, List<Document> documentList) {
    return ListView.builder(
        itemCount: documentList.length,
        itemBuilder: (context, index) {
          return _buildMySpaceListItem(context, documentList[index]);
        });
  }
```
This way we can avoid user to `SwipeToRefresh` when MySpace is in `LoadingState`

Reply:
> This way we can avoid user to SwipeToRefresh when MySpace is in LoadingState

You mean the loading progress is showed in LoadingState?

Reply:
I mean, currently while in `LoadingState` we still be able to swipe to refresh. Thats should not happen.

Reply:
```
          return mySpaceState.viewState.fold(
              (failure) => _buildMySpaceListView(context, mySpaceState.documentList),
              (success) =>
                  success is LoadingState ?
                  _buildMySpaceListView(context, mySpaceState.documentList) :
                  RefreshIndicator(
                      onRefresh: () async => mySpaceViewModel.getAllDocument(),
                      child: _buildMySpaceListView(context, mySpaceState.documentList)));
```

Got it. Please move it in other method

Reply:
You are right, it's good

Reply:
I have to include `_buildUploadFileHere` too

Reply:
Calling 3 times `_buildMySpaceListView` bothers me a lot :(

Reply:
I checked with the case `GetAllDocumentInteractor` return `MySpaceFailure`. Screen is empty instead of showing the `Upload your files here` text. Please fix that.

Reply:
Fixed it, is it ok for you ?

Reply:
Great!